lib/cublas/linalg.jl

```diff
@@ -339,6 +347,48 @@ end
 
 # triangular
 
+if VERSION >= v"1.10-"
+LinearAlgebra.generic_trimatmul!(C::DenseCuMatrix{T}, uploc, isunitc, tfun::Function, A::DenseCuMatrix{T}, B::DenseCuMatrix{T}) where {T<:CublasFloat} =
+    trmm!('L', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), A, B, C)
+LinearAlgebra.generic_mattrimul!(C::DenseCuMatrix{T}, uploc, isunitc, tfun::Function, A::DenseCuMatrix{T}, B::DenseCuMatrix{T}) where {T<:CublasFloat} =
+    trmm!('R', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), B, A, C)
+# tri-tri-mul!
+function LinearAlgebra.generic_trimatmul!(C::DenseCuMatrix{T}, uplocA, isunitcA, tfunA::Function, A::DenseCuMatrix{T}, B::LinearAlgebra.UpperOrLowerTriangular{T,<:DenseCuMatrix}) where {T<:CublasFloat}
```

Comment:
This function is meant to handle the upper-lower triangular (and vice versa) products. It may be a bit more general in that not both factors need to be non-"unitary", but only one of the factors (the one that is `tri[u/l]`ed). We need to think about what should happen if none of the conditions is met, i.e., either throw or fall back to some generic matmatmul or so.